### PR TITLE
fix(types): correct TypeCast's Next callback to return unknown

### PR DESCRIPTION
**Pull request by bbugh**
Fixes "typeCast Next type appears to be incorrect" #3122

Also replaced references to the old incorrect type signature in referencing comments

### CODE REVIEW

File: website/test/fixtures/external-code-embed/QueryOptions.txt [hunk 1]
```diff
@@ -32,7 +32,7 @@ export interface QueryOptions {
    * Determines if column values should be converted to native JavaScript types. It is not recommended (and may go away / change in the future)
    * to disable type casting, but you can currently do so on either the connection or query level. (Default: true)
    *
-   * You can also specify a function (field: any, next: () => void) => {} to do the type casting yourself.
+   * You can also specify a function (field: any, next: () => unknown) => {} to do the type casting yourself.
```
Comment (wellwelwel) [inline website/test/fixtures/external-code-embed/QueryOptions.txt:35]: There is no need to change the test fixtures or resources in `./website`. They just ensure string extractions in the `ExternalCodeEmbed` tests and are for internal use only 🙋🏻‍♂️

File: website/test/resources/external-code-embed/random-methods.txt [hunk 2]
```diff
@@ -41,7 +41,7 @@ export interface QueryOptions {
    * Determines if column values should be converted to native JavaScript types. It is not recommended (and may go away / change in the future)
    * to disable type casting, but you can currently do so on either the connection or query level. (Default: true)
    *
-   * You can also specify a function (field: any, next: () => void) => {} to do the type casting yourself.
+   * You can also specify a function (field: any, next: () => unknown) => {} to do the type casting yourself.
```
Comment (wellwelwel) [inline website/test/resources/external-code-embed/random-methods.txt:44]: Same as above.

**Review by wellwelwel — CHANGES_REQUESTED** [review]
Thanks, @bbugh.

I've suggested some rollback changes to the website's test files 🤝

**Comment by bbugh** [comment]
@wellwelwel I reverted the fixture files, it appears to be passing the checks now. Sorry for the trouble!

**Review by wellwelwel — APPROVED** [review]

**Comment by wellwelwel** [comment]
Thanks again, @bbugh 🙋🏻‍♂️

**Pull request merged**